vllm/config.py

```diff
@@ -1135,15 +1135,23 @@ def _maybe_override_draft_max_model_len(
     @staticmethod
     def create_draft_parallel_config(
         target_parallel_config: ParallelConfig,
-        speculative_draft_tensor_parallel_size: Optional[int]
+        speculative_draft_tensor_parallel_size: Optional[int],
+        draft_hf_config: PretrainedConfig,
     ) -> ParallelConfig:
         """Create a parallel config for use by the draft worker.
 
         This is mostly a copy of the target parallel config, except the tp_size.
         """
         if speculative_draft_tensor_parallel_size is None:
-            speculative_draft_tensor_parallel_size = \
-                  target_parallel_config.tensor_parallel_size
```

Comment:
Looking at this logic now, I don't think it makes sense since below we explicitly say that `speculative_draft_tensor_parallel_size` cannot be set to any value other than 1, but we default it to be the `target_parallel_config.tensor_parallel_size` which can be greater than 1?

Reply:
Should we just default it to 1 in all cases instead?

Reply:
I think for regular draft models it might currently work either with tp=1 or tp same as target model. That could be the reason for this logic (but it strangely still doesn't allow setting tp to the same as the target model explicitly).